tests/lib/hardware/openstack_libcloud.py

```diff
@@ -233,6 +233,18 @@ def get_connection(self):
         )
         return connection
 
+    def _get_image(self, identifier=None):
+        try:
+            return self._get_image_by_id(identifier)
+        except libcloud.common.exceptions.BaseHTTPError:
+            return self._get_image_by_name(identifier)
```

Comment:
Maybe a debug log line that the image wasn't discovered by ID and are reverting to a full search by name.

Reply:
Done

Reply:
Thanks. But to keep the indentation we should restart the string. For example:
```
         logger.debug('No image found by id.'
                      ' Falling back to search by name')
```